pkg/loop/big.go

```diff
@@ -0,0 +1,30 @@
+package loop
+
+import "math/big"
+
+// BigToBytes returns an encoded [*big.Int] or nil when nil.
+// The encoding uses [*big.Int.Bytes] and a leading sign byte.
+func BigToBytes(b *big.Int) []byte {
+	if b == nil {
+		return nil
+	}
+	var neg byte
+	if b.Sign() < 0 {
+		neg = 1
+	}
+	return append([]byte{neg}, b.Bytes()...)
```

Comment:
I thought I would be able to leverage something pre-existing, but since I didn't, maybe it would be cleaner to just use a proper proto message for these... :thinking: